src/generators/go/renderers/EnumRenderer.ts

```diff
@@ -0,0 +1,44 @@
+import { GoRenderer } from '../GoRenderer';
+import { EnumPreset} from '../../../models';
+import { FormatHelpers } from '../../../helpers';
+import { CommonModel } from '../../../models';
+
+/**
+ * Renderer for Go's `enum` type
+ * 
+ * @extends GoRenderer
+ */
+export class EnumRenderer extends GoRenderer {
+  public defaultSelf(): string {
+    const formattedName = this.model.$id && FormatHelpers.toPascalCase(this.model.$id);
+    const type = this.enumType(this.model);
+    const doc = formattedName && this.renderCommentForEnumType(formattedName, type);
+
+    return `${doc}
+type ${formattedName} ${type}`;
+  }
+
+  enumType(model: CommonModel): string {
+    if (this.model.type === undefined || Array.isArray(this.model.type)) {
+      return 'interface{}';
+    }
+
+    return this.toGoType(this.model.type, model);
+  }
+
+  renderCommentForEnumType(name: string, type: string): string {
+    const globalType = type === 'interface{}' ? 'mixed types' : type;
+    return this.renderComments(`${name} represents an enum of ${globalType}.`);
+  }
+
+  renderComments(lines: string | string[]): string {
+    lines = FormatHelpers.breakLines(lines);
+    return lines.map(line => `// ${line}`).join('\n');
+  }
```

Comment:
This function should be written in `GoRenderer` class.

Reply:
done